Src/CSharpier.VSCode/README.md

```diff
@@ -39,6 +39,28 @@ You can turn on format-on-save on a per-language basis by scoping the setting:
 }
 ```
 
+### Devcontainers
+
+CSharpier supports DevContainers if it is installed as a local dotnet tool.
+```bash
+# if no .config/dotnet-tools.json file exists
+dotnet new tool-manifest
+
+# add csharpier to manifest
+dotnet tool install csharpier
+
+# rebuild container image
+```
+
 ## Limitations
 
 Format Selection is not supported.
+
+Only `"editor.formatOnSaveMode" : "file"` is supported. If using other modes, you can set `file` by scoping the setting:
+```json
+"editor.formatOnSave": true,
+"editor.formatOnSaveMode": "modifications"
+"[csharp]": {
+    "editor.formatOnSaveMode": "file"
```

Comment:
Is there a way to specify in the extension manifest/code that format mode should always be file?

Reply:
The extension only calls `registerDocumentFormattingEditProvider` but not `provideDocumentRangeFormattingEdits`. Which I think is essentially saying it only supports a whole file edit. Apparently setting `formatOnSaveMode` to `modifications` can still cause problems though.